neal/sampler.py

```diff
@@ -219,58 +206,50 @@ def sample(self, bqm, beta_range=None, num_reads=None, num_sweeps=1000,
             True
 
         """
+        # get the original vartype so we can return consistently
+        original_vartype = bqm.vartype
 
-        if 'sweeps' in kwargs:
-            warnings.warn("The 'sweeps' parameter is deprecated in "
-                          "favor of 'num_sweeps'.", DeprecationWarning)
-            num_sweeps = kwargs.get('sweeps', num_sweeps)
-
-        # if already index-labelled, just continue
-        if all(v in bqm.linear for v in range(len(bqm))):
-            _bqm = bqm
-            use_label_map = False
-        else:
-            try:
-                inverse_mapping = dict(enumerate(sorted(bqm.linear)))
-            except TypeError:
-                # in python3 unlike types cannot be sorted
-                inverse_mapping = dict(enumerate(bqm.linear))
-            mapping = {v: i for i, v in iteritems(inverse_mapping)}
-
-            _bqm = bqm.relabel_variables(mapping, inplace=False)
-            use_label_map = True
-
-        # beta_range, num_sweeps are handled by simulated_annealing
+        # convert to spin (if needed)
+        if bqm.vartype is not dimod.SPIN:
+            bqm = bqm.change_vartype(dimod.SPIN, inplace=False)
 
-        if not (seed is None or isinstance(seed, Integral)):
+        # parse_initial_states could handle seed generation, but because we're
+        # sharing it with the SA algo, we handle it out here
+        if seed is None:
+            seed = randint(0, (1 << 32 - 1))
+        elif not isinstance(seed, Integral):
             raise TypeError("'seed' should be None or a positive integer")
-        if isinstance(seed, Integral) and not (0 < seed < (2**64 - 1)):
+        if not (0 < seed < (2**32 - 1)):
```

Comment:
It bothers me that I made than an `if` rather than an `elif`. But is it worth the extra commit... :laughing:

Reply:
But does it bother you more than all the superfluous parentheses? :laughing: